language-support/hs/bindings/src/DA/Ledger/Types.hs

```diff
@@ -272,6 +277,11 @@ newtype DaysSinceEpoch = DaysSinceEpoch { unDaysSinceEpoch :: Int}
 newtype TemplateId = TemplateId Identifier deriving (Eq,Ord,Show)
 
 newtype ApplicationId = ApplicationId { unApplicationId :: Text } deriving (Eq,Ord,Show)
+
+instance Read ApplicationId where
```

Comment:
Not a huge fan of this Read instance, can we just wrap it in the `ApplicationId` constructor at the call site?

Reply:
What is wrong with it ?

Reply:
`Read` isn’t really great for general purpose parsing in general and more often than not you are better off writing parser functions. In this particular case it just accepts literally every string and wraps in in the constructor. That just seems like it’s asking for bugs. I’m not asking for validation but applying the constructor explicitly at least makes it clearer what is happening.

Reply:
Done.